src/assign_inliner.cpp

```diff
@@ -29,7 +36,14 @@ std::unique_ptr<Index> IndexBlacklister::visit(std::unique_ptr<Index> node) {
 
 std::unique_ptr<Identifier> IndexBlacklister::visit(
     std::unique_ptr<Identifier> node) {
-  if (this->inside_index) this->wire_blacklist.insert(node->value);
+  if (this->inside_index) {
```

Comment:
nit: this and above code looks the same? can be made common?

Reply:
Curse of the copy paste xD, refactored to use a base class with common logic.